File: app/src/main/java/net/squanchy/remoteconfig/RemoteConfigModule.java
package net.squanchy.remoteconfig;

import com.google.firebase.remoteconfig.FirebaseRemoteConfig;

import net.squanchy.BuildConfig;
import net.squanchy.R;

import dagger.Module;
import dagger.Provides;

@Module
public class RemoteConfigModule {

    @Provides
    FirebaseRemoteConfig FirebaseRemoteConfig() {
        return FirebaseRemoteConfig.getInstance();
    }

    @Provides
    RemoteConfig remoteConfig(FirebaseRemoteConfig firebaseRemoteConfig) {
        return new RemoteConfigBuilder(firebaseRemoteConfig)
                .withDebugMode(BuildConfig.DEBUG)
                .withDefaults(R.xml.remote_config_defaults)
                .build();
    }
}

File: app/src/main/java/net/squanchy/remoteconfig/RemoteConfigModule.kt
package net.squanchy.remoteconfig

import com.google.firebase.remoteconfig.FirebaseRemoteConfig
import dagger.Module
import dagger.Provides
import net.squanchy.BuildConfig
import net.squanchy.R

@Module
class RemoteConfigModule {

    @Provides
    fun firebaseRemoteConfig() = FirebaseRemoteConfig.getInstance()

    @Provides
    fun remoteConfig(firebaseRemoteConfig: FirebaseRemoteConfig): RemoteConfig {
        return RemoteConfigBuilder(firebaseRemoteConfig)
            .withDebugMode(BuildConfig.DEBUG)
            .withDefaults(R.xml.remote_config_defaults)
            .build()
    }

    @Provides
    fun featureFlags(remoteConfig: RemoteConfig) = FeatureFlags(remoteConfig)
}
